### PR TITLE
Added new contact form UI

### LINKED ISSUE #5: Create a new contact UI

### PR DESCRIPTION
**Pull request by soorajbhaskaran**
Fixes #5

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Contacts/Pane/Create.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+
+import { Pane, Typography } from "neetoui";
+
+import Form from "./Form";
+
+import { CONTACTS_INITIAL_FORM_VALUES } from "../constants";
+
+const NewContactPane = ({ showPane, setShowPane, setContacts }) => {
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Pane/Create.jsx]: What's the advantage of having the component name as `NewContactPane` and the filename as an unrelated `Create.jsx`?

File: app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import { Button, Pane, Toastr } from "neetoui";
+import { Input, Select } from "neetoui/formik";
+
+import {
+  CONTACTS_FORM_VALIDATION_SCHEMA,
+  ROLES,
+  PROFILE_PICTURE_URL,
+} from "../constants";
+
+const ContactForm = ({ onClose, contact, setContacts }) => {
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx]: Same question as in last comment.

File: app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    setContacts(contacts => [
+      {
+        ...values,
+        id: contacts.length + 1,
+        role: values.role.label,
+        imageUrl: PROFILE_PICTURE_URL,
+        createdAt: new Date(),
+      },
+      ...contacts,
+    ]);
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx]: What is the reason for setting the new contact as the first added contact and not as the last added contact? Meaning here you are always adding the new contact to index 0. Why so? What's the advantage?

File: app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    onClose();
+  };
+
+  return (
+    <Formik
+      initialValues={contact}
+      validateOnBlur={submitted}
+      validateOnChange={submitted}
+      validationSchema={CONTACTS_FORM_VALIDATION_SCHEMA}
+      onSubmit={values => handleSubmit(values)}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx]: Use shorthand whenever possible.
```suggestion
      onSubmit={handleSubmit}
```

File: app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+              name="email"
+              placeholder="Enter your email address"
+            />
+            <Select
+              className="w-full flex-grow-0"
+              label="Role"
+              name="role"
+              options={ROLES}
+              placeholder="Select Role"
+            />
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx]: Why is the role not required?

File: app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                label="Last Name"
+                name="lastName"
+                placeholder="Enter Last Name"
+              />
+            </div>
+            <Input
+              className="w-full flex-grow-0"
+              label="Email Address"
+              name="email"
+              placeholder="Enter your email address"
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx]: Don't we require any of these fields?

File: app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+              style="primary"
+              type="submit"
+              onClick={() => setSubmitted(true)}
+            />
+            <Button
+              label="Cancel"
+              size="large"
+              style="text"
+              onClick={onClose}
+            />
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx]: Go through the Slack thread.

File: app/javascript/src/components/Dashboard/Contacts/constants.js [hunk 8]
```diff
@@ -1,3 +1,32 @@
...
+  lastName: "",
+  email: "",
+  role: null,
+};
+
+export const ROLES = [
+  { label: "Owner", value: "owner" },
+  { label: "Admin", value: "admin" },
+  { label: "Member", value: "member" },
+];
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/constants.js]: Go through the Slack thread. Create a util to build this.

**Review by yedhink — CHANGES_REQUESTED** [review]
@soorajbhskrn

File: app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx (hunk 3 above)
Comment (soorajbhaskaran) [reply app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx]: This was made so for the development easiness. If this was not done then I need to scroll down after creating each contact to verify whether it worked or not.
On the contrary, ideally in the real world, I think the opposite is true.
I will fix this in next commit.

**Pull request merged**